gtsfm/frontend/matcher/superglue_matcher.py

```diff
@@ -0,0 +1,108 @@
+"""SuperGlue matcher implementation
+
+The network was proposed in 'SuperGlue: Learning Feature Matching with Graph Neural Networks' and is implemented by 
+wrapping over author's source-code.
+
+Note: the pretrained model only supports SuperPoint detections currently.
+
+References:
+- http://openaccess.thecvf.com/content_CVPR_2020/papers/Sarlin_SuperGlue_Learning_Feature_Matching_With_Graph_Neural_Networks_CVPR_2020_paper.pdf # noqa
+- https://github.com/magicleap/SuperGluePretrainedNetwork
+
+Authors: Ayush Baid, John Lambert
+"""
+from typing import Tuple
+
+import numpy as np
+import torch
+
+from gtsfm.common.keypoints import Keypoints
+from gtsfm.frontend.matcher.matcher_base import MatcherBase
+from thirdparty.SuperGluePretrainedNetwork.models.superglue import SuperGlue
+
+SUPERGLUE_DESC_DIM = 256
+
+
+class SuperGlueMatcher(MatcherBase):
+    """Implements the SuperGlue matcher -- a pretrained graph neural network using attention."""
+
+    def __init__(self, use_cuda: bool = True, use_outdoor_model: bool = True):
+        """Initialize the configuration and the parameters."""
+        super().__init__()
+
+        self._config = {
+            "descriptor_dim": 256,
+            "weights": "outdoor" if use_outdoor_model else "indoor",
+            "sinkhorn_iterations": 20,
```

Comment:
Maybe make it variable too. Also comment that we are using Author's params.

Reply:
good catch -- updated